vizro-core/src/vizro/models/_navigation/_accordion.py

```diff
@@ -84,32 +75,24 @@ def _create_default_accordion(self):
             id=f"{self.id}_outer",
         )
 
-    def _create_accordion(self):
-        accordion_items = []
+    def _create_default_accordion(self):
+        """Creates a default accordion with all pages provided to the Dashboard."""
+        registered_pages = [page for page in dash.page_registry.keys() if page != MODULE_PAGE_404]
```

Comment:
Does it make sense to define `self.pages` in `Accordion._pre_build` method (based on custom `self.pages` or `dash.page_registry` for default accordion creation) and then to reuse the `self.pages` in methods:
 - _create_default_accordion,
 - _create_custom_accordion,
 - _create_accordion_buttons,

In that way we don't need to calculate pages (from `dash.page_registry.keys()`) each time the navigation is built (and it's built every time the page is changed).

Also, then we don't need to propagate pages object to the `_create_accordion_buttons` every time.

Reply:
I like the push! It gave me another idea, so I am deviating a bit from your proposal, but you got me inspired 😄 

I've moved the default page call with `dash.page_registry.keys()` to the validator `_validate_pages`. Having it in the validator instead of the pre-build has the advantage of re-using that validator in the Navigation model and other subsequent navigation-type models, so we're not repeating the same code in several pre-builds. It also makes a pre-build method redundant.

Because of that change, we can eliminate the `_create_default_accordion` private method!

Reply:
Happy to see latest changes 😄.